lib/droplet_kit/resources/kubernetes_clusters_resource.rb

```diff
@@ -0,0 +1,64 @@
+module DropletKit
+  class KubernetesClustersResource < ResourceKit::Resource
+    include ErrorHandlingResourcable
+
+    resources do
+      action :all, 'GET /v2/kubernetes/clusters' do
+        query_keys :per_page, :page, :tag_name
+        handler(200) { |response| KubernetesClusterMapping.extract_collection(response.body, :read) }
+      end
+
+      action :find, 'GET /v2/kubernetes/clusters/:id' do
+        handler(200) { |response| KubernetesClusterMapping.extract_single(response.body, :read) }
+      end
+
+      action :create, 'POST /v2/kubernetes/clusters' do
+        body { |object| KubernetesClusterMapping.representation_for(:create, object) }
+        handler(201) { |response, cluster| KubernetesClusterMapping.extract_into_object(cluster, response.body, :read) }
+        handler(422) { |response| ErrorMapping.fail_with(FailedCreate, response.body) }
+      end
+
+      action :config, 'GET /v2/kubernetes/clusters/:id/kubeconfig' do
+        handler(200) { |response| response.body }
+      end
+
+      action :update, 'PUT /v2/kubernetes/clusters/:id' do
+        body { |cluster| KubernetesClusterMapping.representation_for(:update, cluster) }
+        handler(202) { |response| KubernetesClusterMapping.extract_single(response.body, :read) }
+        handler(422) { |response| ErrorMapping.fail_with(FailedUpdate, response.body) }
+      end
+
+      action :upgrade, 'GET /v2/kubernetes/clusters/:cluster_id/upgrade' do
```

Comment:
Let's remove the upgrade endpoint, we're not sure how this will look just yet.

Reply:
sounds good.